components/automate-ui/src/app/components/settings-sidebar/settings-sidebar.component.spec.ts

```diff
@@ -64,46 +64,53 @@ describe('SettingsSidebarComponent', () => {
   describe('IAM v2', () => {
     beforeEach(() => {
       component.iamMajorVersion$ = observableOf(<IAMMajorVersion>'v2');
-      component.iamMinorVersion$ = observableOf(<IAMMinorVersion>'v1');
     });
 
-    it('shows all links consistent with settings-landing', () => {
-      fixture.detectChanges();
-      const links = element.nativeElement
-        .querySelectorAll('div.nav-items chef-sidebar-entry');
-      expect(links.length).toBe(settingsLandingComponent.routeList.length);
-    });
+    using([
+      [false, 9, 'v2'],
+      [true, 10, 'v2.1']
+    ], function (projectsEnabled: boolean, linkCount: number, versionName: string) {
 
-    it('has route order consistent with settings-landing', () => {
-      fixture.detectChanges();
-      const links = element.nativeElement
-        .querySelectorAll('div.nav-items chef-sidebar-entry');
-      for (let i = 0; i < settingsLandingComponent.routeList.length; i++) {
-        expect(links[i].getAttribute('route'))
-          .toBe(settingsLandingComponent.routeList[i].route);
-      }
-    });
+      it(`shows all links consistent with settings-landing for ${versionName}`, () => {
+        component.projectsEnabled$ = observableOf(projectsEnabled);
+        fixture.detectChanges();
+        const links = element.nativeElement
+          .querySelectorAll('div.nav-items chef-sidebar-entry');
+        expect(links.length).toBe(linkCount);
+      });
+
+      it(`has route order consistent with settings-landing for ${versionName}`, () => {
+        component.projectsEnabled$ = observableOf(projectsEnabled);
+        fixture.detectChanges();
+        const links = element.nativeElement
+          .querySelectorAll('div.nav-items chef-sidebar-entry');
+        for (let i = 0; i < links.length; i++) {
+          expect(links[i].getAttribute('route'))
+            .toBe(settingsLandingComponent.routeList[i].route);
+        }
+      });
 
-   it('has paths consistent with settings-landing', () => {
-     fixture.detectChanges();
-     const elements = Array
-       .from<HTMLElement>(
-         element.nativeElement.querySelectorAll('div.nav-items app-authorized'))
-       .filter(elem =>
-         elem.firstElementChild && elem.firstElementChild.tagName === 'CHEF-SIDEBAR-ENTRY');
+      it(`has paths consistent with settings-landing for ${versionName}`, () => {
+        component.projectsEnabled$ = observableOf(projectsEnabled);
+        fixture.detectChanges();
+        const elements = Array
+          .from<HTMLElement>(
```

Comment:
💡 Interesting. Didn't know [`Array.from`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from)

Reply:
Particularly helpful here because, though we already have a list, it is not quite an array, so this just massages it a bit.